src/components/timeline/TimelineCursors.svelte

```diff
@@ -0,0 +1,133 @@
+<script lang="ts">
+  import type { ScaleTime } from 'd3-scale';
+  import { getDoyTime } from '../../utilities/time';
+
+  export let mouseOver: MouseOver;
+  export let xScaleView: ScaleTime<number, number> | null = null;
+  export let marginLeft: number = 50;
+  export let drawWidth: number = 0;
+  export let cursorHeaderHeight: number = 20;
+  export let cursorEnabled: boolean = true;
+
+  $: onMouseOver(mouseOver);
+  $: onCursorEnableChange(cursorEnabled);
+
+  let raf: number = null;
+  let cursorDiv: HTMLElement = null;
+  let cursorLabelDiv: HTMLElement = null;
+  let offsetX: number = -1;
+
+  function onMouseOver(event: MouseOver | undefined) {
+    if (event && xScaleView && cursorDiv && cursorLabelDiv) {
+      offsetX = event.e.offsetX;
+      updateCursor();
+    }
+  }
+
+  function onCursorEnableChange(cursorEnabled) {
+    if (cursorEnabled) {
+      updateCursor();
+    } else {
+      hideCursor();
+    }
+  }
+
+  function updateCursor() {
+    if (cursorEnabled && offsetX >= 0 && offsetX <= drawWidth) {
+      const unixEpochTime = xScaleView.invert(offsetX).getTime();
+      const doyTime = getDoyTime(new Date(unixEpochTime));
+
+      cancelAnimationFrame(raf);
+      raf = window.requestAnimationFrame(() => {
```

Comment:
Is the `requestAnimationFrame` necessary here? My reading of this is the cursor should be updated on each `mouseOver`, which gets called on mouse pointer change.

Reply:
It's maybe not "necessary", but improves FPS. `mouseOver` events will be fired async, and it's unknown at which point in the render cycle we'll respond to it. Theoretically multiple `mouseOver` events could happen in the span of one frame. Using `rAF` ensures we only tell the browser to update the cursor position when it's ready to paint, and makes sure we only ever update once per frame.

Reply:
Ah nice good to know. So you're optimizing the update. I *think* Svelte will take care of this since it only updates during the next microtask. So changes to `mouseOver` will be batched and the update will only happen once during the next microtask. See: https://lihautan.com/compile-svelte-in-your-head-part-2/#schedule-update
I'm fine with keeping this for now tho since I don't think it hurts anything.